src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/VerifyClientDeviceIdentityTest.java

```diff
@@ -0,0 +1,181 @@
+                "BrokerSubscribingToCertUpdates")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+
+            VerifyClientDeviceIdentityRequest request = new VerifyClientDeviceIdentityRequest()
+                    .withCredential(new ClientDeviceCredential()
+                            .withClientDeviceCertificate("VALID PEM"));
+
+            Pair<CompletableFuture<Void>, Consumer<VerifyClientDeviceIdentityResponse>> cb = asyncAssertOnConsumer((m) -> {
+                assertTrue(m.isIsValidClientDevice()) ;
+            });
+
+            verifyClientIdentity(ipcClient, request, cb.getRight());
+            cb.getLeft().get(10, TimeUnit.SECONDS);
+        }
+    }
+
+    @Test
+    void GIVEN_broker_WHEN_verify_client_identity_with_invalid_pem_THEN_verify() throws Exception {
+        kernel.getContext().put(IotAuthClient.class, iotAuthClient);
+        startNucleusWithConfig("cda.yaml");
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "BrokerSubscribingToCertUpdates")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+
+            VerifyClientDeviceIdentityRequest request = new VerifyClientDeviceIdentityRequest()
+                    .withCredential(new ClientDeviceCredential()
+                            .withClientDeviceCertificate("INVALID PEM"));
+
+            Pair<CompletableFuture<Void>, Consumer<VerifyClientDeviceIdentityResponse>> cb = asyncAssertOnConsumer((m) -> {
+                assertFalse(m.isIsValidClientDevice());
+
+            });
+
+            verifyClientIdentity(ipcClient, request, cb.getRight());
+            cb.getLeft().get(10, TimeUnit.SECONDS);
+        }
+    }
+
+    @Test
+    void GIVEN_broker_with_no_config_WHEN_verify_client_identity_THEN_error_is_thrown()
+            throws ExecutionException, InterruptedException {
+        kernel.getContext().put(IotAuthClient.class, iotAuthClient);
+        startNucleusWithConfig("BrokerNotAuthorized.yaml");
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "BrokerWithNoConfig")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+            VerifyClientDeviceIdentityRequest request = new VerifyClientDeviceIdentityRequest()
+                    .withCredential(new ClientDeviceCredential()
+                            .withClientDeviceCertificate("VALID PEM"));
+            Pair<CompletableFuture<Void>, Consumer<VerifyClientDeviceIdentityResponse>> cb =
+                    asyncAssertOnConsumer((m) -> {
+                    });
+            Exception err = Assertions.assertThrows(Exception.class, () -> {
+                verifyClientIdentity(ipcClient, request, cb.getRight());
+            });
+            assertThat(err.getCause(), is(instanceOf(UnauthorizedError.class)));
```

Comment:
you have to use `cb.getLeft().get` to ensure it was called and only called the expected number of times. If you aren't doing this, then there really isn't any point in using `asyncAssertOnConsumer`